### PR TITLE
Pull Request for popup and homepage window, with TvMaze and Involvement API in ShowAddict Capstone

### PR DESCRIPTION
**Pull request by Fombi-Favour**
In this Capstone project,
## Highlights 💯
**Group Task**
- Setting up the project with `webpack` and `Jest`
- Added the header
- Added the footer
- Finding the external API
- Choose assets

**Student A: @Fombi-Favour Task**
- Display number of likes for each item on the Homepage
- Add all items counter on the homepage
- Create feature: Add new like
- Display list of items on the homepage
- Add test for items counter

**Student B: @zuhebahmed88091 Task**
- Display comments for a given item on the Comments pop-up
- Add comments counter
- Create feature: Add new comment
- Display Comments pop-up with selected item's details
- Add test for comments counter

## 🚀 Live demo
[Check the live demo. Click here!](https://showaddict-fombi-zuheb.netlify.app/)

## 🎥 Presentation
Check out the presentation for this project. [Click here!](https://drive.google.com/file/d/1YQWbMSfqdHp7QGdG7ui-9hOLhkjXKIwF/view?usp=sharing)

## 👨 Authors
- **[Zuheb Ahmed](https://github.com/zuhebahmed88091)**
- **[Fombi Favour](https://github.com/Fombi-Favour)**

### CODE REVIEW

File: src/modules/commentCount.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* eslint import/prefer-default-export: "off" */
+import { commentUrl } from './api.js';
+
+const testComments = async (id) => {
+  const commenturl = `?item_id=${id}`;
+  const response = await fetch(commentUrl + commenturl);
+  const result = await response.json();
+  return result;
+};
```
Comment (gemmen29) [inline src/modules/commentCount.js:9]: - Kindly make sure to follow the guidelines in the card `Add comments counter - Student B` in this [issues](https://github.com/zuhebahmed88091/ShowAddit_capstone_js/issues/5): 
   - [ ] A counter function should look for specific DOM elements. 
   - [ ] A counter function should cover all the edge cases you can think about.

File: src/modules/itemCount.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { seriesApiUrl } from './api.js';
+
+const itemCounter = (item) => {
+  const totalCount = document.getElementById('total-count');
+  totalCount.innerHTML = `(${item})`;
+};
```
Comment (gemmen29) [inline src/modules/itemCount.js:6]: - Kindly make sure to follow the guidelines in the card `Add all items counter on the Homepage - Student A` in this [issues](https://github.com/zuhebahmed88091/ShowAddit_capstone_js/issues/6): 
   - [ ] A counter function should look for specific DOM elements. 
   - [ ] A counter function should cover all the edge cases you can think about.

File: src/__test__/itemCount.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          type: 'Scripted',
+          language: 'English',
+        },
+      },
+    ],
+  ),
+}));
+describe('items counter tests using Jest', () => {
+  test('items count ', async () => {
+    const response = await testShows();
```
Comment (gemmen29) [inline src/__test__/itemCount.test.js:29]: - [ ] Kindly update the test for the count of all items based on the new implementation and please make sure that your test includes all edge cases like if there are 0 items on the page and use the `itemCounter` method that is used to count items instead of `testShows` method.

File: src/__test__/commentCount.test.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { testComments } from '../modules/commentCount.js';
```
Comment (gemmen29) [inline src/__test__/commentCount.test.js:1]: - [ ] Kindly update the test for the count of all comments based on the new implementation and please make sure that your test includes all edge cases like if there are 0 items on the page.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by medaminedev66** [comment]
Hello @zuhebahmed88091 👋,

_This is an additional review as a result of the second opinion request._

I'm not able to find the counter functions that look for specific DOM elements (e.g. for the comments counter it should look for comments) and make the counting based on what is actually displayed on the page. The desired functions should return the number of items or comments. In the modal.js file, I found only a function responsible for displaying a modal using the DOM.
According to the requirements, even if the API gives you these numbers, you should create a specific function to calculate these numbers in each page.

Regarding the tests, if you create those functions, it is important to include unit tests to ensure their functionality. You should consider various scenarios, including cases where there are no items on the page (in which the counter function should return 0). By incorporating these scenarios into your tests, you can ensure their strength and reliability

Keep rocking 🚀!

**Review by julie-ify — APPROVED** [review]
Hi @Fombi-Favour and @zuhebahmed88091,

## Highlights!
- Counter tests ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
